[WIP] Build all Native Modules with Zig CC, eliminating NativePath

jazzay opened the pull request:
# PR Details

This is a WIP with the goal to build all Native modules with Zig CC via this nice Nuget package:
https://github.com/vezel-dev/zig-toolsets

The Zig toolchain wraps a modern clang and also bundles full cross plat std libs, so we no longer need most of NativePath.

For now I have brought some minimal pieces of NativePath into Stride.Native, and will revisit the locations of this common code once I get through the other Native modules. Possibly Stride.Native can remain the shared native source location.

Note currently Zig cross compiler does not support iOS/Android although iOS is almost ready and sort of works. In the short term we may need to maintain a target dedicated to these Mobile platforms.

I have tested the builds on both Windows and Mac across all our desktop platforms. See Stride.Native.csproj.

Seeking feedback on this approach prior to finishing all aspects.

Remaining tasks:

- [ ] Update Stride.Audio
- [ ] Update Stride.Navigation
- [ ] Update Stride.VirtualReality
- [ ] Figure out solution for iOS
- [ ] Figure out solution for Android
- [ ] Ensure full stride packaging, etc
- [ ] Test Editor and with sample Game

## Related Issue
https://github.com/stride3d/stride/issues/360

https://github.com/stride3d/stride/pull/1757

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] My change requires a change to the documentation.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

arturoc committed:
openxr: remove custom code from game

not needed by merging commit and flush on OpenXrHmd

arturoc committed:
remove unused variable

arturoc committed:
openxr: dispose

arturoc committed:
openxr: Remove mirror texture check from forward renderer

Instead check it on the settings by using a CanMirrorTexture flag that VRDevices can report back if they don't support texture mirroring

ericwj committed:
Migrate remaining uses of Stride.Core.Interop.

arturoc committed:
Fix wrong formatted curly brace

ericwj committed:
Merge branch 'stride3d:master' into MigrateInterop3

ericwj committed:
Merge branch 'stride3d:master' into MigrateInterop2

ericwj committed:
Delete `CoreUtilities` and most of `StrideCoreExtensions`.

ericwj committed:
Merge branch 'MigrateInterop2' of https://github.com/ericwj/stride into MigrateInterop2

ericwj committed:
Merge.

ericwj committed:
Fix compiler errors after merge.

ericwj committed:
Fix compiler errors after merge.

ericwj committed:
Merge branch 'MigrateInterop2' of https://github.com/ericwj/stride into MigrateInterop2

ericwj committed:
Use `is`. Fix `#define CHECKED`.

ericwj committed:
Use `is`. Fix `#define CHECKED`.

ericwj committed:
Merge branch 'MigrateInterop2' of https://github.com/ericwj/stride into MigrateInterop2

ericwj committed:
Use is. Fix #define CHECKED.

ericwj committed:
Merge branch 'MigrateInterop2' of https://github.com/ericwj/stride into MigrateInterop2

ericwj committed:
Fix Vulkan build.

ericwj committed:
Review fixed statements for null and access size.

ericwj committed:
Review fixed statements for null and access size.

ericwj committed:
Merge amend

ericwj committed:
Delete unfinished work in progress.

ericwj committed:
Fix confusion of size with alignment.

ericwj committed:
Bring back `void Interop.Pin<T>(T)` until `UpdateEngine` is redesigned.

ericwj committed:
Rigorously check `ParameterCollection.CopyTo<T>` bounds.

ericwj committed:
Fix `Stride.Rendering.ParameterCollection` buffer allocation size off-by-one alignment padding.

ericwj committed:
Fix Android build.

ericwj committed:
Fix Android build.

ericwj committed:
Change the diagnostic code for UpdateEngine unsafe operations.

ericwj committed:
Implement Stream.Read/Write(Span) and Serialize(Span)

ericwj committed:
Implement Stream.Read/Write(Span) and Serialize(Span)

ericwj committed:
Implement Stream.Read/Write(Span) and Serialize(Span)

ericwj committed:
Remove NativeStream, NativeMemoryStream from inheritance graph.
Remove Read/Write(nint, int), also from SerializationStream.
Polyfill with extension methods. Source compatible; binary breaking.

ericwj committed:
Provide obsoletion messages in BinarySerialization.
Proper exceptions in new methods; fix messages in others.

ericwj committed:
Remove duplicate.

ericwj committed:
Delay initializing `out` variables in `GetHeadPoseUnsafe`.

ericwj committed:
Fix asset compiler build.

ericwj committed:
Fix access violation in particle system simulation processor.

ericwj committed:
Correct exception behavior of derived `System.IO.Stream` types.
Remove manually copied documentation comments in favor of `<inheritdoc/>`.

ericwj committed:
Do not use trailing alignment padding for source parameter collection buffer.
Fixes pre-existing buffer overrun and those introduced by assuming trailing alignment padding.

ericwj committed:
Fix whitespace in LZ4Stream.cs

ericwj committed:
Rewrite parameter buffer copying.
To account for differences in type/size of equally named parameters.

ericwj committed:
`StreamExtensions` should have been public.

Basewq committed:
[Samples] Fix TopDownRPG punch animation caused by breaking change in raycast from stride3d#1105

ericwj committed:
Fix debug assertion error in LZ4Stream.Write.

allcontributors[bot] committed:
docs: update README.md [skip ci]

allcontributors[bot] committed:
docs: update .all-contributorsrc [skip ci]

allcontributors[bot] committed:
docs: update README.md [skip ci]

allcontributors[bot] committed:
docs: update .all-contributorsrc [skip ci]

ericwj committed:
Typo.

ericwj committed:
Obsolete `SerializationStream.NativeStream` to change the name.

VaclavElias committed:
Readme: Build Instructions improvements (#1494)

ericwj committed:
Comments and whitespace.

palkercode committed:
Added overload for Entity constructor

insomnyawolf committed:
[Debug] DebugTextSystem configurable time on-screen (#1515)

garychia committed:
Enable multiple selections of the axes of grids to be drawn

Basewq committed:
[Build] Add build support to referencing projects that use package version with Semantic Versioning 2.0.0 format

Basewq committed:
[Math] Fix Matrix.Invert to return zero matrix if inversion fails instead of NaNs, caused by breaking change from (#1301) (#1520)

Co-authored-by: Basewq <Basewq@users.noreply.github.com>

allcontributors[bot] committed:
docs: update README.md [skip ci]

allcontributors[bot] committed:
docs: update .all-contributorsrc [skip ci]

manio143 committed:
Merge pull request #1521 from Basewq/semver-build-support

Add build support to referencing projects that use SemVer 2.0.0 format

palkercode committed:
Merge branch 'stride3d:master' into master

arturoc committed:
openxr: Add static extensionsvariable

To allow to set extensions from outside the engine

Basewq committed:
[Math] Fix Matrix.Decompose, Matrix.DecomposeXYZ & Quaternion.RotationYawPitchRoll to better handle the edge cases where the pitch (or yaw) is +/-90 degrees

Basewq committed:
[Editor] Fix rotation value showing -0 due to MathF respecting negative sign even at zero

manio143 committed:
Configure Samples.Tests and SamplesTestServer to auto resolve packages correctly

manio143 committed:
Merge pull request #1535 from manio143/SamplesTestsAutoResolver

Fix Samples.Tests and SamplesTestServer to auto-resolve packages

Eideren committed:
[Template] Fix first and third person delta time usage (#1511)

Eideren committed:
[Template] FPS, remove unused logic (#1512)

Eideren committed:
Addressing PVS report

Eideren committed:
Strip encoding issues

Eideren committed:
Replace encoding issue

manio143 committed:
Improve documentation of ListFiles and add tests for DatabaseFileProvider implementation

manio143 committed:
Replace string concatenation with interpolation

manio143 committed:
Merge pull request #1547 from manio143/dbFileProviderSearchDocumentation

Improve documentation of ListFiles and add tests for DatabaseFileProvider implementation

arturoc committed:
OpenXR: Fix Non DirectX11 platforms

manio143 committed:
Merge pull request #1469 from ericwj/MigrateInterop2

Improve memory management, embrace modern .NET

Jklawreszuk committed:
Replace native ThreadSleep with .NET Thread.Sleep

manio143 committed:
Merge pull request #1523 from Jklawreszuk/replace-native-sleep-with-net-thread-sleep

Replace native ThreadSleep with .NET Thread.Sleep

Ethereal77 committed:
Use Path.Combine() to concatenate MSBuild paths

manio143 committed:
Merge pull request #1569 from Ethereal77/fix-msbuild-path-combine

Use Path.Combine() to concatenate MSBuild paths

arturoc committed:
Merge branch 'openxr-dx11-master' of github.com:arturoc/stride into openxr-dx11-master

arturoc committed:
VRRendererSettings: Remove CopyMirror property code. Not really useful since this can only be properly evaluated at runtime

arturoc committed:
Update Silk OpenXR nuget

arturoc committed:
VirtualReality project: change OpenXR Silk to 2.10.1 to match other Silk packages in the rest of the engine

manio143 committed:
Merge pull request #1545 from Eideren/addressing_pvs_report

[Misc] Addressing PVS report

dependabot[bot] committed:
Bump System.Drawing.Common from 4.7.0 to 4.7.2 in /sources/engine/Stride

Bumps [System.Drawing.Common](https://github.com/dotnet/corefx) from 4.7.0 to 4.7.2.
- [Release notes](https://github.com/dotnet/corefx/releases)
- [Commits](https://github.com/dotnet/corefx/commits)

---
updated-dependencies:
- dependency-name: System.Drawing.Common
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

manio143 committed:
Merge pull request #1560 from stride3d/dependabot/nuget/sources/engine/Stride/System.Drawing.Common-4.7.2

Bump System.Drawing.Common from 4.7.0 to 4.7.2 in /sources/engine/Stride

palkercode committed:
Merge branch 'stride3d:master' into master

garychia committed:
Simplify code

palkercode committed:
Update Entity.cs

Updated with Eideren's suggestion

manio143 committed:
Merge pull request #1499 from palkercode/master

Added overload for Entity constructor

Brian-Hamlin-BCS committed:
Updated Entity.cs

Fixed an error preventing the solution from compiling due to Entity.cs.

manio143 committed:
Merge pull request #1578 from Brian-Hamlin-BCS/master

Updated Entity.cs

skohanowski committed:
Added deconstructors to most of the structs in Stride.Mathematics

skohanowski committed:
Addressed @manio143 feedback regarding deconstructors.

manio143 committed:
Merge pull request #1582 from skohanowski/Deconstructors

Added deconstructors to most of the structs in Stride.Mathematics

manio143 committed:
Make a few changes around viewport grid axis rotation and hiding

manio143 committed:
Introduce an enum to avoid allocating the nullable object

garychia committed:
Fix the default scene settings data

mattiascibien committed:
Fix documentation error on Int2

manio143 committed:
Merge pull request #1585 from mattiascibien/fix_int2_docs

Fix documentation error on Int2

Jklawreszuk committed:
Use LZ4 as external dependency

manio143 committed:
Merge pull request #1566 from Jklawreszuk/lz4-package

Use LZ4 as external dependency

Jklawreszuk committed:
Remove settings.json and update gitignore

manio143 committed:
Merge pull request #1586 from Jklawreszuk/vscode-gitignore

Remove settings.json and update gitignore

Eideren committed:
[Rendering] Addressing PVS report - SDL (#1546)

Eideren committed:
[Physics] Update WorldMatrix when forcing transformation on PhysicsComponent (#1406)

xen2 committed:
Merge remote-tracking branch 'arturoc/openxr-dx11-master' (closes #1483)

mattiascibien committed:
Fix #1466: adding scripts by dragging

Jklawreszuk committed:
Cleanup Windows Phone refs

Jklawreszuk committed:
Cleanup Silverlight refs

Jklawreszuk committed:
Cleanup WinStore refs

Jklawreszuk committed:
Cleanup Win8 & Win10 refs

Jklawreszuk committed:
Cleanup NET Portable,MicroFramework,Client

Jklawreszuk committed:
Remove WinStore content for VRSandbox sample

azeno committed:
[Graphics] If possible use flip model for swap chain

https://devblogs.microsoft.com/directx/dxgi-flip-model

Jklawreszuk committed:
Move launcher prerequisites validation to separate file

xen2 committed:
Keep local copy of signclient to avoid issues with teamcity

Jklawreszuk committed:
Launcher - Remove unnecessary namespaces, format and simplify code

manio143 committed:
Merge pull request #1605 from Jklawreszuk/launcher-refactor

[Launcher] Code refactor

garychia committed:
Merge pull request #1 from manio143/viewportgrid_changes

Viewportgrid changes

manio143 committed:
Merge pull request #1516 from garychia/GridsInMultiplePlanes

Enable multiple selections of the axes of grids to be drawn

manio143 committed:
Document what's in the Core folder

manio143 committed:
Merge pull request #1610 from manio143/document_core_folder

[Docs] Add a Readme with information on what's in the Core folder

manio143 committed:
Merge pull request #1588 from mattiascibien/fix_adding_scripts_by_dragging

Fix #1466: adding scripts by dragging

manio143 committed:
Update NuGet version to cover most .NET 6 SDK users for MSBuild issue

manio143 committed:
Add debug information & remove legacy code

manio143 committed:
Merge pull request #1613 from manio143/fix_script_editor

Update NuGet dependencies to 6.4.0 (temporary fix for ScriptEditor issues)

dependabot[bot] committed:
Bump SSH.NET from 2016.1.0 to 2020.0.2 in /sources/engine/Stride.Assets

Bumps [SSH.NET](https://github.com/sshnet/SSH.NET) from 2016.1.0 to 2020.0.2.
- [Release notes](https://github.com/sshnet/SSH.NET/releases)
- [Commits](https://github.com/sshnet/SSH.NET/compare/2016.1.0...2020.0.2)

---
updated-dependencies:
- dependency-name: SSH.NET
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

Doprez committed:
fixed constructors in CsharpBeginner template

manio143 committed:
Merge pull request #1618 from Doprez/fix-begginner-errors

fixed constructors in CsharpBeginner template

manio143 committed:
Merge pull request #1614 from stride3d/dependabot/nuget/sources/engine/Stride.Assets/SSH.NET-2020.0.2

Bump SSH.NET from 2016.1.0 to 2020.0.2 in /sources/engine/Stride.Assets

manio143 committed:
Merge pull request #1428 from vvvv/dev/azeno/native-libs-use-runtimes-folder

[Build] Native libraries get copied to runtimes/RID/native folder

azeno committed:
[Graphics] Fixes Vsync off when using new flip model

azeno committed:
[Build] Fixes native libraries not copied to output folder of unit tests

For example the Stride.Navigation.Tests didn't contain the necessary libnavigation.dll in its runtimes folder.

xen2 committed:
Merge pull request #1632 from stride3d/dev/azeno/unit-test-native-lib-load-fix

[Build] Fixes native libraries not copied to output folder of unit tests

azeno committed:
[Graphics] Adds try/catch blocks to flip model and tearing support queries to support older OS

azeno committed:
[Graphics] Fixes crash when flip model is not supported

The `AllowTearing` flag must not be set in that case

Doprez committed:
rebased branch

manio143 committed:
Merge pull request #1643 from Doprez/change-msbuild

rebased branch

manio143 committed:
Merge pull request #1591 from Jklawreszuk/cleanup-legacy-platforms

Cleanup legacy platforms

NicusorN5 committed:
Possible fix for the animation importer crashing

`FbxTime::GetOneFrameValue(FbxTime::eFrames1000);` returns the smallest value, so there's the highest animation resolution possible(?)

Jklawreszuk committed:
[Core] Remove some obsolete code

NicusorN5 committed:
Update AnimationConverter.h

NicusorN5 committed:
Also check negative values

manio143 committed:
Merge pull request #1653 from NicusorN5/master

Possible fix for the animation importer crashing

NicusorN5 committed:
Update the forum link with Github discussions link

Doprez committed:
added null check

manio143 committed:
Merge pull request #1656 from Doprez/fix-NativeDevice-causing-crash

fix Crash when NativeDeviceContext is null

azeno committed:
[Build] Fixes location of native openxr_loader.dll in Stride.VirtualReality package as well as used Silk version

This broke with the merge of PR #1428 because these lines of code where added after #1428 was created.

xen2 committed:
Merge pull request #1660 from vvvv/fix-openxr-load-location

[Build] Fixes location of native openxr_loader.dll in Stride.VirtualReality package

manio143 committed:
Merge pull request #1654 from Jklawreszuk/core-obsoletes

[Core] Remove some obsolete code

Jklawreszuk committed:
Fix XML warnings (#1678)

Jklawreszuk committed:
[deps] - Remove unused ATITC folder (#1668)

Jklawreszuk committed:
[deps] - Remove unused HtmlRenderer (#1670)

Jklawreszuk committed:
[deps] - Remove unused ExpresionBlend Sdk (#1669)

Jklawreszuk committed:
Remove VisualStudio reattach deadcode (#1677)

Jklawreszuk committed:
Fix XML documentation (#1680)

IXLLEGACYIXL committed:
fix argument reference

IXLLEGACYIXL committed:
remvoe reference to non existing class that doesnt get referenced

IXLLEGACYIXL committed:
fix light to renderlight

IXLLEGACYIXL committed:
blend layer context is the only class that actually sets and evaluates the enum values

IXLLEGACYIXL committed:
remove duplicate through inheritdoc

IXLLEGACYIXL committed:
add missing inheritdoc

IXLLEGACYIXL committed:
add missing inheritdoc

IXLLEGACYIXL committed:
fix wrong reference to draweffect parameters

IXLLEGACYIXL committed:
Merge branch 'FixXmlDocs' of https://github.com/NexStandard/stride into FixXmlDocs

IXLLEGACYIXL committed:
fix wrong comment. False class in  comment

IXLLEGACYIXL committed:
duplicate comment

IXLLEGACYIXL committed:
duplicate comment

IXLLEGACYIXL committed:
duplicate comment

IXLLEGACYIXL committed:
fix wrong interface link

IXLLEGACYIXL committed:
remove duplicate

IXLLEGACYIXL committed:
fix wrong reference

IXLLEGACYIXL committed:
fix resourcegrouplayout reference

IXLLEGACYIXL committed:
fix vertex comment

IXLLEGACYIXL committed:
wrong param name, its still a weird name as it was previous a sorter i assume.

IXLLEGACYIXL committed:
ParticleSorter is now ParticleList from previous interface renaming

manio143 committed:
Merge pull request #1682 from NexStandard/FixXmlDocs

[Docs] Fix xml docs

IXLLEGACYIXL committed:
[Docs] Fix xml docs (#1683)

Ethereal77 committed:
Renamed build scripts in solution files and removed missing ones (#1696)

azeno committed:
[Reflection] Fixes loading of mixed mode assemblies in .NET Core (#1691)

It seems .NET Core behaves slightly different in this regard, throwing a different kind of exception, so previous code no longer applies, even though it's the same assembly being loaded (irrKlang.NET4.dll)

Jklawreszuk committed:
[deps] - Remove unused SSH (#1687)

Jklawreszuk committed:
[deps] Remove unused libgit2 dll's (#1686)

Jklawreszuk committed:
[deps] - Remove unused Irony (#1685)

Eideren committed:
Show useful error message and directions to users lacking requirements for Stride.Core.Tasks

Eideren committed:
Merge remote-tracking branch 'origin/master' into tasks_clarification

Eideren committed:
Revert "Merge remote-tracking branch 'origin/master' into tasks_clarification" (#1701)

This reverts commit 06dc98ab5e742d9ac34c5422403e25cd278e2e00.

Jklawreszuk committed:
Remove native code from stride.core project (#1688)

* Remove native code from stride.core project

Eideren committed:
Show useful error message and directions to users lacking requirements for Stride.Core.Tasks

Eideren committed:
Add dot between major and minor

Jklawreszuk committed:
Separate VisualStudio projects from main solution (#1684)

Jklawreszuk committed:
Fix edge case collision behavior with BoxColliderShape by creating Box2D shape when is2D (#1708)

IXLLEGACYIXL committed:
[GameStudio] Fix sub projects reloading - changing the comparison project related and not UPath related (#1704)

* fix creation and renaming for sub projects

bro-these-are-all-taken committed:
[Rendering] Vulkan Fullscreen Fix (#1709)

* Enabling swapchain recreation in SwapChainGraphicsPresenter.Vulkan.cs so that changes in resolution at runtime don't stop the game from rendering to the screen.

IXLLEGACYIXL committed:
[WIP] Replace NativeStream with BinaryWriter (#1710)

* Rename false Comments of Nativestream to Stream

* Replace any calls to BinaryReader/BinaryWriter

* Remove native stream IO

* remove using directives of deleted IO namespace

* Remove obsolete comment

* replace nativestream comment and block writing with copyto as it allready exists

Ethereal77 committed:
[Rendering] Fix missing Utilities.CopyMemory for other graphics APIs (#1697)

* Fix Direct3D 12 uses of Utilities.CopyMemory

* Also some minor cleanup

* Fix OpenGL uses of Utilities.CopyMemory

* Also some minor cleanup

* Fix Vulkan uses of Utilities.CopyMemory

* Also some minor cleanup

* Add a section to Stride.props where users can set design-time graphics API

* Fix more usages of CopyMemory and ClearMemory

* Also remove some helper methods that copied memory in favor of new built-in ones

* Comment modifications based on feedback

* Fix a deserialization issue introduced in commit 189e96980e96d6180e18a8f7e9466945ff9ecbe8

Eideren committed:
Docs: Readme, update Community Projects link (#1720)

allcontributors[bot] committed:
docs: update README.md [skip ci]

allcontributors[bot] committed:
docs: update .all-contributorsrc [skip ci]

allcontributors[bot] committed:
docs: add Jklawreszuk as a contributor for code (#1727)

* docs: update README.md [skip ci]

* docs: update .all-contributorsrc [skip ci]

---------

Co-authored-by: allcontributors[bot] <46447321+allcontributors[bot]@users.noreply.github.com>

Jklawreszuk committed:
Change Assimp to Silk.NET.Assimp implementation (#1719)

* Changed Assimp binding to Silk.Net.Assimp

* Converted Stride.Importer.Assimp to a C# library and changed assimp binding to Silk.Net.Assimp
* Converted Stride.Importer.Common to a C# library as this did not contain any native code

allcontributors[bot] committed:
docs: add Ethereal77 as a contributor for code (#1729)

allcontributors[bot] committed:
docs: add tamamutu as a contributor for code (#1731)

allcontributors[bot] committed:
docs: add IXLLEGACYIXL as a contributor for code (#1732)

allcontributors[bot] committed:
docs: add arturoc as a contributor for code (#1733)

allcontributors[bot] committed:
docs: add garychia as a contributor for code (#1734)

allcontributors[bot] committed:
docs: add NicusorN5 as a contributor for code (#1735)

allcontributors[bot] committed:
docs: add mattiascibien as a contributor for code (#1736)

Jklawreszuk committed:
Update README badges (#1737)

Jklawreszuk committed:
[deps] - Remove unused assimp source code

Eideren committed:
[Samples] Bump samples version (#1756)

VaclavElias committed:
Readme.md update - Improving readme based on the community feedback (#1739)

* Removed duplicated badge

* Missing lfs added

arturoc committed:
Fix DX11 shared textures

Right now creating a texture from a native texture with D3D11_RESOURCE_MISC_SHARED_NTHANDLE  will fail unless it is also D3D11_RESOURCE_MISC_SHARED_KEYEDMUTEX. This combination is recomended but not mandatory acording to the specification: https://learn.microsoft.com/en-us/windows/win32/api/d3d11/ne-d3d11-d3d11_resource_misc_flag

Eideren committed:
[System] Fix binary path location when the dlls are run from an unexpected executable

xen2 committed:
Fix VSIX build

xen2 committed:
Assimp: preload assimp library from Ultz.Native.Assimp so that it can be found when actually needed (fixes #1765)

xen2 committed:
Removed UWP from README.md

VaclavElias committed:
[Service] Xml comments improvements (#1768)

Aggror committed:
Update README.md

I build stride source from scratch today but building worked only partially.

I was have SDK 7 installed, but was still missing SDK 6.0.

Also the Git LFS clone command is deprecated and buildin into Git clone

tebjan committed:
[Shader] Adds sRGB and Gamma conversions to ColorUtility.sdsl (#1744)

Co-authored-by: tebjan <tebjan@gmail.com>

Basewq committed:
[Graphics] Fix expected image size for OpenGL textures. (#1786)

Co-authored-by: Basewq <Basewq@users.noreply.github.com>

manio143 committed:
[Profiler] Refactor Profiler and submit metrics to .NET Diagnostics API (#1774)

Eideren committed:
[Asset] Show warning message when importing assets that are not inside of the resource folder (#1781)

manio143 committed:
Merge pull request #1746 from Jklawreszuk/assimp-code

[deps] - Remove unused assimp source code

manio143 committed:
Merge pull request #1702 from Eideren/clarification_2

[Build] Show useful error message and directions for users lacking requirements to run Stride.Core.Tasks

Eideren committed:
[Project] Show custom message for users with outdated dotnet version (#1767)

Eideren committed:
[Editor] PropertyGrid, re-implement member sorting by declaration/token (#1790)

Doprez committed:
[Sample] Fix SpriteFont sample, typo in service fetch (#1798)

SVNMLR committed:
[Editor] Add dynamic snapping (#1801)

Implements a dynamic snapping used while holding down a key on manipulating (rotating/moving/scaling) an object/entity

chkw-rks committed:
[Readme] Update logo for dark and light theme (#1587)

Jklawreszuk committed:
[deps] - Remove unused WindowsAPICodePack (#1747)

IXLLEGACYIXL committed:
[Editor] Repair NPE introduced in #1781 (#1804)

Co-authored-by: IXLLEGACYIXL <ixllegacy123@outlook.com>

tebjan committed:
Adds Awesome Stride to readme

jazzay committed:
Stride.Native builds with Zig CC and no NativePath

VaclavElias committed:
Update README.md - Bullet point fixed in Prerequisites (#1808)

xen2 commented:
Looks great!
Small detail: I think your indentation settings for csproj doesn't match what we usually have (2 spaces, which is what Visual Studio generates). This results in diff being harder to visualize.

jazzay commented:
> Small detail: I think your indentation settings for csproj doesn't match 

I will adjust my formatting to match the Stride style.

cNoNim committed:
[Editor] Fix some xaml errors (#1822)

Eideren committed:
[Graphics] Dx11: Fix ResizeBuffer when swap chain is under flip model and changing format (#1814)

VaclavElias committed:
[Core] Basic support for F# and VB Project Types (#1821)

SeleDreams committed:
[Editor] Added the ability to copy imported assets automatically to the Resources dir (#1827)

Doprez committed:
[Samples] Add more documentation for beginner tutorial (#1830)

MeharDT committed:
[Editor] Check to see if the .NET 6.0 SDK is Installed When Launching Stride (#1809)

Jklawreszuk committed:
Remove unused MonoDevelop.Debugger (#1748)

Jklawreszuk committed:
[deps] - Remove unused monolinker (#1749)

SVNMLR committed:
[Editor] Add new Theme & UI fixes/refinements (#1807)

MeharDT committed:
[iOS] .NET 6 Template (#1833)

- Updated iOS template .csproj to support .NET Core/6.0, Stride will no longer throw an error when generating iOS projects

jazzay committed:
Merge branch 'stride3d:master' into native-xplat
